src/confidential/elip151.rs

```diff
@@ -29,20 +29,12 @@ use crate::descriptor::{DescriptorSecretKey, SinglePriv};
 use crate::extensions::{Extension, ParseableExt};
 use crate::{Descriptor as OrdinaryDescriptor, DescriptorPublicKey, Error};
 
-/// The SHA-256 initial midstate value for the [`Elip151Hash`].
-const MIDSTATE_ELIP151: [u8; 32] = [
-    0x49, 0x81, 0x61, 0xd8, 0x52, 0x45, 0xf7, 0xaa, 0xd8, 0x24, 0x27, 0xb5, 0x64, 0x69, 0xe7, 0xd6,
-    0x98, 0x17, 0xeb, 0x0f, 0x27, 0x14, 0x6f, 0x4e, 0x7b, 0x95, 0xb3, 0x6e, 0x46, 0xc1, 0xb5, 0x61,
-];
-
-sha256t_hash_newtype!(
-    Elip151Hash,
-    Elip151Tag,
-    MIDSTATE_ELIP151,
-    64,
-    doc = "ELIP-151 Deterministic descriptor blinding keys",
-    forward
-);
+sha256t_hash_newtype! {
+    pub struct Elip151Tag = hash_str("ELIP-151 Deterministic descriptor blinding keys");
```

Comment:
🐞🐞🐞

sorry :(

Reply:
Lol, derp. But we really should have had tests to catch this!!

Reply:
would it be feasible and makes sense to have a parameter of the macro requiring to specify the result of the hash for the empty string and the macro generates a test for it?

Reply:
oo I like that idea.

cc @tcharding what do you think?

Reply:
I don't quite get it, what is the test hoping to catch? Is it hoped to be a regression test? (I don't see how one would get the hash of the empty string without first using the new tagged type.) Am I missing some context?

Reply:
Yes, you'd run the test once to get the correct string.

And then yes, it would be a regression test which would (hopefully) detect whenever anybody poked at the macro, or replaced the macro, or something, causing the hash to change.

Having said that, having such a test as *part of the macro* limits its value, since it's likely to go away when the macro is changed, which is exactly when we want it..

Reply:
We could definitely, and probably should already, have a single unit test in each repo that calls the macro and tests the empty hash - "don't trust verify" and all that.

Reply:
I hear those lads that maintain `hashes` chop and change every week and are a bunch of cowboys.